From: Jiri Pirko <jiri@mellanox.com>
Date: Wed, 9 Aug 2017 14:30:32 +0200
Subject: net: sched: propagate classid down to offload drivers
Patch-mainline: v4.14-rc1
Git-commit: 7690f2a51d8afe51ac97e7fae66b081f192a7158
References: bsc#1056787

Drivers need classid to decide they support this specific qdisc+class
or not. So propagate it down via the tc_cls_common_offload struct.

Signed-off-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/net/pkt_cls.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/net/pkt_cls.h
+++ b/include/net/pkt_cls.h
@@ -464,6 +464,7 @@ struct tc_cls_common_offload {
 	u32 chain_index;
 	__be16 protocol;
 	u32 prio;
+	u32 classid;
 };
 
 static inline void
@@ -474,6 +475,7 @@ tc_cls_common_offload_init(struct tc_cls
 	cls_common->chain_index = tp->chain->index;
 	cls_common->protocol = tp->protocol;
 	cls_common->prio = tp->prio;
+	cls_common->classid = tp->classid;
 }
 
 struct tc_cls_u32_knode {
